lib/hanami/rspec/generators/support_db_cleaning.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require "database_cleaner/sequel"
+
+# Clean the databases between tests tagged as `:db`
+RSpec.configure do |config|
+  all_databases = -> {
+    slices = [Hanami.app] + Hanami.app.slices.with_nested
+
+    slices.each_with_object([]) { |slice, dbs|
+      next unless slice.key?("db.rom")
+
+      dbs.concat slice["db.rom"].gateways.values.map(&:connection)
+    }.uniq
+  }
```

Comment:
I made this a proc because if I instead made it a method, then I'd need a place to _put_ it. In my own apps I use `module Test` (and various nested modules within it) to put utility code like this, but for standard Hanami apps, I don't think we're yet at the point where we can introduce a structure like that. I'd like to do it, but it will need a bit more thought on the approach, as well as additional effort put into docs, both of which too much to fit in our remaining time before 2.2.


And so we have the proc. It remains local, so no concrete method naming required, and given the size of this file, I think it's still readable enough.

Reply:
I agree with your caution. While RSpec is clearly the default in the community, it would be good to be forward-thinking and allow something new and interesting to be used in addition to just minitest, and that flexibility will be hard.

What concerns me about RSpec is how difficult parallelism is with it, which contributes to long-term slow test suites in my experience. The modular separation of Slices in Hanami would otherwise make parallel testing very attractive.

Reply:
Cool, thanks for considering this, @alassek. Glad to hear you agree.

(I'd love for us to provide a parallel testing experience in the future!)